src/hotspot/share/prims/jvmtiEnv.cpp

```diff
@@ -3210,7 +3212,10 @@ JvmtiEnv::GetMethodName(Method* method, char** name_ptr, char** signature_ptr, c
 jvmtiError
 JvmtiEnv::GetMethodDeclaringClass(Method* method, jclass* declaring_class_ptr) {
   NULL_CHECK(method, JVMTI_ERROR_INVALID_METHODID);
-  (*declaring_class_ptr) = get_jni_class_non_null(method->method_holder());
+  Klass* k = method->method_holder();
+  Handle holder(Thread::current(), k->klass_holder()); // keep the klass alive
```

Comment:
How do we know `k` is not already unloaded?

Reply:
We don't know `k` would be unloaded or not in the middle of GC concurrent marking. So I think we need to keep it alive just like accessing weak reference.

Reply:
As @fisk mentioned above, `k->is_loader_alive()` is checked in caller `jvmti_GetMethodDeclaringClass`. So we won't enter here if k is already unloaded.  The exception this PR handles is to deal with the `k` in the middle of concurrent marking that `is_loader_alive` returns `true` but could be unloaded soon. We'd better keep it alive just like accessing weak reference.